### ISSUE TITLE
Version 0.27.0-insiders3 refers to invalid path and fails installing on MacOS

### ISSUE THREAD

**Issue by X-Ryl669** [microsoft/vscode-cpptools#5146]
Type: General
I've tried to update the extension with Version 0.27.0-insiders3: March 13, 2020 but it failed installing with these errors:
```
Warning : Expected file ~/.vscode-oss/extensions/ms-vscode.cpptools-0.27.0-insiders3/debugAdapters/lldb/bin/debugserver is missing.

Warning : Expected file ~/.vscode-oss/extensions/ms-vscode.cpptools-0.27.0-insiders3/debugAdapters/lldb/bin/lldb-mi is missing.
Warning : Expected file ~/.vscode-oss/extensions/ms-vscode.cpptools-0.27.0-insiders3/debugAdapters/lldb/bin/lldb-argdumper is missing.

Warning : Expected file ~/.vscode-oss/extensions/ms-vscode.cpptools-0.27.0-insiders3/debugAdapters/lldb/bin/lldb-launcher is missing.

Dependencies installation finished

Installation of C/C++ extension failed. For more information, refer to Output window.
```

(sorry, I've translated the message in english).
Here's what I have in the referred folder:

```
$ pwd
~/.vscode-oss/extensions/ms-vscode.cpptools-0.27.0-insiders3/debugAdapters
$ find .
.
./OpenDebugAD7
./bin
./bin/Microsoft.MICore.XmlSerializers.dll
./bin/Microsoft.DebugEngineHost.dll
./bin/Microsoft.VisualStudio.Shared.VSCodeDebugProtocol.dll
./bin/Microsoft.MICore.dll
./bin/osxlaunchhelper.scpt
./bin/Microsoft.VisualStudio.Debugger.Interop.11.0.dll
./bin/Newtonsoft.Json.dll
./bin/cppdbg.ad7Engine.json
./bin/Microsoft.VisualStudio.Debugger.Interop.15.0.dll
./bin/Microsoft.VisualStudio.Debugger.Interop.12.0.dll
./bin/Microsoft.VisualStudio.Debugger.InteropA.dll
./bin/Microsoft.VisualStudio.Debugger.Interop.10.0.dll
./bin/Microsoft.MIDebugEngine.dll
./bin/WindowsDebugLauncher.exe
./bin/OpenDebugAD7.exe
./framework
./framework/Microsoft.CSharp.dll
./framework/System.Xml.Linq.dll
./framework/Mono.Posix.dll
./framework/System.Configuration.dll
./framework/config.osx
./framework/System.Runtime.Serialization.dll
./framework/config.linux
./framework/mscorlib.dll
./framework/Mono.CSharp.dll
./framework/System.ServiceModel.Internals.dll
./framework/Facades
./framework/Facades/System.IO.dll
./framework/Facades/System.Dynamic.Runtime.dll
./framework/Facades/System.Reflection.dll
./framework/Facades/System.Threading.dll
./framework/Facades/System.Text.Encoding.Extensions.dll
./framework/Facades/System.Linq.Expressions.dll
./framework/Facades/System.Runtime.InteropServices.dll
./framework/Facades/System.Text.RegularExpressions.dll
./framework/Facades/System.ObjectModel.dll
./framework/Facades/System.Xml.XDocument.dll
./framework/Facades/System.Collections.dll
./framework/Facades/System.Xml.ReaderWriter.dll
./framework/Facades/System.Threading.Tasks.dll
./framework/Facades/System.Text.Encoding.dll
./framework/Facades/System.Globalization.dll
./framework/Facades/System.Runtime.dll
./framework/Facades/System.Reflection.Extensions.dll
./framework/Facades/System.Runtime.Extensions.dll
./framework/Facades/System.Linq.dll
./framework/Facades/System.Diagnostics.Tools.dll
./framework/Facades/System.Diagnostics.Debug.dll
./framework/Facades/System.Runtime.Serialization.Primitives.dll
./framework/Mono.Security.dll
./framework/System.dll
./framework/System.Xml.dll
./framework/System.ComponentModel.Composition.dll
./framework/System.Security.dll
./framework/System.Core.dll
./lldb-mi
./lldb-mi/bin
./lldb-mi/bin/lldb-mi
./lldb-mi/NOTICE.TXT
./mono.osx
```
As you can see, `lldb-mi` is not in `./bin/lldb-mi` but `./lldb-mi/bin/lldb-mi`
I don't have `./bin/lldb` folder at all.

**Describe the bug**
- OS and Version: MacOS 10.13.6 (High Sierra)
- VS Code Version: 1.34.0
- C/C++ Extension Version: 0.27.0-insiders3
- Other extensions you installed (and if the issue persists after disabling them): Not related
- XCode 10.1 installed
- A clear and concise description of what the bug is.

**To Reproduce**
Install the extension

**Expected behavior**
Installation works

**Comment by sean-mcmanus** [comment]
How are you installing 0.27.0-insiders3? The vsix should contain an install.vsix file that should prevent the downloading you are seeing. Can you uninstall the extension, restart VS Code, verify the extension is gone, and then re-install the 0.27.0-insiders3 vsix or upgrade to it?

**Comment by sean-mcmanus** [comment]
Oh, I get those warning messages too, but I don't get the failure to install message (on Catilena). I think the warning messages are not a problem, but the failure message is.

**Comment by sean-mcmanus** [comment]
Looks like an issue with Mac High Sierra or lower.

**Comment by WardenGnaw** [comment]
This is an issue specifically with an offline installer.

The 0.27.0-insiders3 has the debugger bits prepackaged and it is not downloaded online. There is code to check to see if you are running High Sierra or Lower, or Mojave or Higher in the package download and it will give you the correct files. 

To work around this issue, you can delete the `lldb-mi` folder and get the `lldb` folder from https://go.microsoft.com/fwlink/?LinkID=817244

That folder will need to be placed in the extension root and the files will need to have '755' permissions on them.

**Comment by X-Ryl669** [comment]
I did not use an offline installer (or maybe a long time ago). insiders extension updated itself and failed.

Your step worked.

**Comment by sean-mcmanus** [comment]
@X-Ryl669 When using the updateChannel "Insiders" setting, it automatically downloads/installs the "offline vsix". We have a fix pending at https://github.com/microsoft/vscode-cpptools/pull/5153 .

**Comment by sean-mcmanus** [comment]
FYI, this is fixed with 0.27.0-insiders4.

**Comment by X-Ryl669** [comment]
Thanks!
